rasa/utils/tensorflow/environment.py

```diff
@@ -0,0 +1,129 @@
+import logging
+import os
+from typing import Text, Dict
+import typing
+
+if typing.TYPE_CHECKING:
+    from tensorflow import config as tf_config
+
+import rasa.utils.common as rasa_utils
+from rasa.constants import (
+    ENV_GPU_CONFIG,
+    ENV_CPU_INTER_OP_CONFIG,
+    ENV_CPU_INTRA_OP_CONFIG,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def setup_gpu_environment() -> None:
+    """Set configuration for a GPU environment based on the environment variable set"""
+
+    gpu_memory_config = os.getenv(ENV_GPU_CONFIG)
+    if gpu_memory_config:
+
+        # Import from tensorflow only if necessary(environment variable was set)
+        from tensorflow import config as tf_config
+
+        parsed_gpu_config = parse_gpu_config(gpu_memory_config)
+        physical_gpus = tf_config.list_physical_devices("GPU")
+
+        # Logic taken from https://www.tensorflow.org/guide/gpu
+        if physical_gpus:
+            for gpu_id, gpu_id_memory in parsed_gpu_config.items():
+
+                allocate_gpu_memory(physical_gpus[gpu_id], gpu_id_memory)
+
+        else:
+            rasa_utils.raise_warning(
+                f"You have an environment variable '{ENV_GPU_CONFIG}' set but no GPUs were detected to configure"
+            )
+
+
+def allocate_gpu_memory(
+    gpu_instance: "tf_config.PhysicalDevice", logical_memory: int
+) -> None:
```

Comment:
either add a docstring or make it private. I think all functions in this module can be private except the `setup_tf_environment`, correct?

Reply:
@wochinge Yes made all of them private.